spec/bolt_server/transport_app_spec.rb

```diff
@@ -69,17 +70,52 @@ def file_data(file)
         )
     end
 
+    describe '/plans/:module_name/:plan_name' do
+      let(:path) { "/plans/foo/bar?environment=production" }
+      let(:fake_pal) { instance_double('BoltServer::PE::PAL') }
+
+      it 'returns properly from /plans' do
+        expect(BoltServer::PE::PAL).to receive(:new).and_return(fake_pal)
+        expect(fake_pal).to receive(:get_plan_info).with('foo::bar').and_return('name' => 'foo::bar')
+        get(path)
+        metadata = JSON.parse(last_response.body)
+        expect(metadata).to eq('name' => 'foo::bar')
+      end
+    end
+
     describe '/plans' do
       let(:path) { "/plans?environment=production" }
       let(:fake_pal) { instance_double('BoltServer::PE::PAL') }
 
+      it 'returns properly from /plans' do
+        expect(BoltServer::PE::PAL).to receive(:new).and_return(fake_pal)
+        expect(fake_pal).to receive(:list_plans).and_return([['abc'], ['def']])
+        get(path)
+        metadata = JSON.parse(last_response.body)
+        expect(metadata).to eq([{ 'name' => 'abc' }, { 'name' => 'def' }])
+      end
+
+      it 'returns 404 if an environment not found error is thrown' do
+        # Actually creating the EnvironmentNotFound error with puppet is difficult to do without
+        # puppet actually loaded with settings, so just stub out the error type
+        stub_const("Puppet::Environments::EnvironmentNotFound", StandardError)
+        expect(BoltServer::PE::PAL).to receive(:new).and_raise(Puppet::Environments::EnvironmentNotFound)
+        get(path)
+        expect(last_response.status).to eq(404)
+      end
+    end
+
+    describe '/plans?metadata=true' do
```

Comment:
Can you merge this with `describe '/plans'`?

Reply:
yeah